ui/src/index.ts

```diff
@@ -284,6 +283,10 @@ window.addEventListener("load", async function () {
       changePageTo(noCampaignPage);
       return;
     }
+    if (alreadyVoted) {
+      changePageTo(alreadyVotedPage);
+      return;
+    }
```

Comment:
This should be done right after the API call. It's not useful to setup the home page (lines 176 to 280) if we're not going to display it. Also that makes the use of `campaign` and `alreadyVoted` right next to their declaration, which is better in terms of ease of reading the code.

Reply:
Ok, I had to make a few more changes ensure we could still view the stats page even if there was no campaign or the user already voted. Same for the displayHomeButton (we could otherwise vote by going to the stats and coming back to the home page)